src/Swarm/Game/Display.hs

```diff
@@ -76,20 +85,29 @@ defaultChar :: Lens' Display Char
 --   the 'defaultChar' will be used.
 orientationMap :: Lens' Display (Map Direction Char)
 
+-- | The display caches the current orientation of the entity, so we
+--   know which character to use from the orientation map.
+curOrientation :: Lens' Display (Maybe Direction)
```

Comment:
I am getting very lost here. Do all entities have an orientation map just because robots need it?

And now we need to put here the robot direction because we are at display level?

Reply:
@byorgey I wonder if that is a good responsibility division - perhaps there is a better abstraction/wrapper for displayable things that have orientation, and their drawing depends on it. :thinking: 

I am hoping things would get simpler if robots (or later other entities) set the drawn character from their direction map when they are themselves changing their orientation, keeping the source of truth in the robot (or entity).

Caching seems like overengineering to me - I am worried it would just cause bugs.

Reply:
These are good concerns.  To be honest I'm not super happy with this either but it was the best I could come up with.  I'll try to give it some more thought.

If we have robots/entities set the drawn character from their direction map, that is still caching something --- it's just caching a character instead of a `Direction`.  I don't really see how that would be an improvement.  Notice how I changed the `robotDisplay` to be a special lens that automatically updates the cached direction when returning the `Display`.  So there shouldn't be any way to get a `Display` that is out of sync.  Well, I mean, there is always a way, but it seems like you would have to try pretty hard to turn it into a bug.

I think the direction map should remain part of the `Display` --- it is part of the description of how an entity should be drawn on the screen.

Reply:
Hmm, but maybe it would indeed be better to store the current character in the `Display` instead of a `Direction`.  Then robots and entities could do things like e.g. temporarily change their display by setting the current character to something else?